src/utils/flagging.R

```diff
@@ -173,13 +208,32 @@ calculate_flags <- function(
   # check when we had a day with value above x
   # and 0 in the previous # number of dates
   if (!null_fs) {
-    first_since <- sort(first_since, decreasing = FALSE)
-    for (fs_period in first_since) {
-      .data <- dplyr$mutate(
-        .data,
-        "flag_first_{{ fs_period }}" := .data[[x_col]] > 0 & dplyr$lag(.data[[paste0("rs_", fs_period)]]) == 0
+    # create the forward looking 60 day period for checking the minimums
+    .data <- dplyr$mutate(
+      .data,
+      rs_next_60 = zoo$rollsum(
+        x = .data[[x_col]],
+        k = 60,
+        fill = "extend",
+        align = "left"
       )
-    }
+    )
+
+    df_fs <- purrr$map2(
+      .x = first_since,
+      .y = first_since_minimums,
+      .f = \(fs_period, fs_minimum) {
+        dplyr$mutate(
+          .data,
+          "flag_first_{{ fs_period }}" := .data[[x_col]] > 0 & dplyr$lag(.data[[paste0("rs_", fs_period)]]) == 0 & rs_next_60 >= fs_minimum
```

Comment:
maybe a couple comments here - took a while for me to figure out what was going on